devicetypes/qubino/qubino-din-dimmer.src/qubino-din-dimmer.groovy

```diff
@@ -0,0 +1,722 @@
+		state.currentPreferencesState."$key".status = "synced"
+	} else if (preference.type == "boolRange") {
+		if (state.currentPreferencesState."$key".status == "disablePending" && preferenceValue == preference.disableValue) {
+			state.currentPreferencesState."$key".status = "disabled"
+		} else {
+			runIn(5, "syncConfiguration", [overwrite: true])
+		}
+	} else {
+		state.currentPreferencesState."$key"?.status = "syncPending"
+		runIn(5, "syncConfiguration", [overwrite: true])
+	}
+	// Preferences template end
+}
+
+private getPreferenceValue(preference, value = "default") {
+	def integerValue = value == "default" ? preference.defaultValue : value.intValue()
+	switch (preference.type) {
+		case "enum":
+			return String.valueOf(integerValue)
+		case "boolean":
+			return String.valueOf(preference.optionActive == integerValue)
+		default:
+			return integerValue
+	}
+}
+
+private getCommandValue(preference) {
+	def parameterKey = preference.key
+	switch (preference.type) {
+		case "boolean":
+			return settings."$parameterKey" == "true" ? preference.optionActive : preference.optionInactive
+		case "boolRange":
+			def parameterKeyBoolean = parameterKey + "Boolean"
+			return !notNullCheck(settings."$parameterKeyBoolean") || settings."$parameterKeyBoolean" ? settings."$parameterKey" : preference.disableValue
+		case "range":
+			return settings."$parameterKey"
+		default:
+			return Integer.parseInt(settings."$parameterKey")
+	}
+}
+
+private notNullCheck(value) {
+	return value != null
+}
+
+private encap(cmd, endpoint = null) {
+	if (cmd) {
+		if (endpoint) {
+			cmd = zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint: endpoint).encapsulate(cmd)
+		}
+
+		if (zwaveInfo.zw.endsWith("s")) {
+			zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+		} else {
+			cmd.format()
+		}
+	}
+}
+
+private isPreferenceChanged(preference) {
+	if (notNullCheck(settings."$preference.key")) {
+		if (preference.type == "boolRange") {
+			def boolName = preference.key + "Boolean"
+			if (state.currentPreferencesState."$preference.key".status == "disabled") {
+				return settings."$boolName"
+			} else {
+				return state.currentPreferencesState."$preference.key".value != settings."$preference.key" || !settings."$boolName"
+			}
+		} else {
+			return state.currentPreferencesState?."$preference.key".value != settings."$preference.key"
+		}
+	} else {
+		return false
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelEndPointReport cmd, ep = null) {
+	log.debug "MultiChannelEndPointReport: ${cmd}"
+	if (!childDevices) {
+		createChildDevice()
+	}
+	response([refresh()])
+}
+
+def createChildDevice() {
+	try {
+		String childDni = "${device.deviceNetworkId}-2"
+		String componentLabel = "Qubino Temperature Sensor"
+		def properties = [
+			completedSetup: true,
+			label         : componentLabel,
+			isComponent   : false
+		]
+		addChildDevice(
+			"Child Temperature Sensor",
```

Comment:
I don't think this DTH is in the qubino namespace.

Reply:
The thing is that the existing Child Temperature Sensor in "smartthings" namespace supports "Battery" capability (although it might have been named in a better way that reflects that, for example: "Child Temperature Sensor Battery"). 
We added a "Child Temperature Sensor" dth in "qubino" namespace (https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/28554) - to avoid renaming/refactoring (and other complications) of an existing:  "Child Temperature Sensor" (the one in "smartthings" namespace).
**Three possible ways here:**
1) Use the Child Temperature Sensor (qubino namespace) as proposed
2) Refactor "Child Temperature Sensor" (smartthings namespace) -> "Child Temperature Sensor Battery" and then move (from qubino) a child DTH to the smartthings namespace: "Child Temperature Sensor" (without support of Battery capability)
3) use "Child Temperature Sensor" only (smartthings namespace) without any refactoring, but we will have to create a fake battery event on initialization, so it would be 100% of battery all the time.
@greens What do You think ?